ui/packages/catalog-ui-search/src/main/webapp/react-component/container/map-actions/map-actions.tsx

```diff
@@ -0,0 +1,132 @@
+        return action.get('id').indexOf('catalog.data.metacard.map.') === 0
+      })
+    }
+
+    getOverlayActions = () => {
+      let modelOverlayActions = this.getActions().filter(function(action: any) {
+        return (
+          action.get('id').indexOf('catalog.data.metacard.map.overlay.') === 0
+        )
+      })
+
+      return _.map(modelOverlayActions, (modelOverlayAction: any) => {
+        return {
+          description: modelOverlayAction.get('description'),
+          url: modelOverlayAction.get('url'),
+          overlayText: this.getOverlayText(modelOverlayAction.get('url')),
+        }
+      })
+    }
+
+    getOverlayText = (actionUrl: String) => {
+      const overlayTransformerPrefix = 'overlay.'
+      const overlayTransformerIndex = actionUrl.lastIndexOf(
+        overlayTransformerPrefix
+      )
+      if (overlayTransformerIndex >= 0) {
+        let overlayName = actionUrl.substr(
+          overlayTransformerIndex + overlayTransformerPrefix.length
+        )
+        return 'Overlay ' + overlayName + ' on the map'
+      }
+
+      return ''
+    }
+
+    overlayImage = (event: any) => {
+      const clickedOverlayUrl = event.target.getAttribute('data-url')
+      const removeOverlay = clickedOverlayUrl === this.state.currentOverlayUrl
+
+      if (removeOverlay) {
+        this.props.model.unset('currentOverlayUrl', { silent: true })
+        this.setState({ currentOverlayUrl: '' })
+        wreqr.vent.trigger(
+          'metacard:overlay:remove',
+          this.props.model.get('metacard').get('id')
+        )
+      } else {
+        this.props.model.set('currentOverlayUrl', clickedOverlayUrl, {
+          silent: true,
+        })
+        this.setState({ currentOverlayUrl: clickedOverlayUrl })
+        this.props.model
+          .get('metacard')
+          .set('currentOverlayUrl', clickedOverlayUrl)
+        wreqr.vent.trigger('metacard:overlay', this.props.model.get('metacard'))
+      }
+      this.render()
+    }
+
+    render() {
+      const { currentOverlayUrl } = this.state
+      return this.getMapActions().length !== 0 ? (
+        <>
+          <div className="is-header">Map:</div>
+          <div className="is-divider" />
+          <div className="actions">
+            {this.getOverlayActions().map((overlayAction: any) => {
+              return (
+                <a
+                  style={{
+                    cursor: 'pointer',
+                  }}
+                  data-url={overlayAction.url}
+                  title={overlayAction.description}
+                  onClick={this.overlayImage}
+                  key={overlayAction.url}
+                >
+                  {overlayAction.overlayText}
+                  {overlayAction.url === currentOverlayUrl ? ' (remove)' : ''}
+                </a>
+              )
+            })}
+          </div>
+          <div className="is-divider" />
+        </>
+      ) : (
+        ''
```

Comment:
I think in React you should normally be returning null as opposed to an empty string.